builder/config.go

```diff
@@ -0,0 +1,358 @@
+	DebugInfoLib    string `required:"false" toml:"DEBUG_INFO_LIB"`
+	DebugInfoSrc    string `required:"false" toml:"DEBUG_INFO_SRC"`
+}
+
+type mixerConf struct {
+	LocalBundleDir string `required:"false" toml:"LOCAL_BUNDLE_DIR"`
+	LocalRepoDir   string `required:"false" toml:"LOCAL_REPO_DIR"`
+	LocalRPMDir    string `required:"false" toml:"LOCAL_RPM_DIR"`
+}
+
+// LoadDefaults sets sane values for the config properties
+func (config *MixConfig) LoadDefaults() error {
+	if !UseNewConfig {
+		return nil
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	// [Builder]
+	config.Builder.BundleDir = filepath.Join(pwd, "mix-bundles")
+	config.Builder.Cert = filepath.Join(pwd, "Swupd_Root.pem")
+	config.Builder.ServerStateDir = filepath.Join(pwd, "update")
+	config.Builder.VersionPath = pwd
+	config.Builder.DNFConf = filepath.Join(pwd, ".yum-mix.conf")
+
+	// [Swupd]
+	config.Swupd.Bundle = "os-core-update"
+	config.Swupd.ContentURL = "<URL where the content will be hosted>"
+	config.Swupd.Format = "1"
+	config.Swupd.VersionURL = "<URL where the version of the mix will be hosted>"
+
+	// [Server]
+	config.Server.DebugInfoBanned = "true"
+	config.Server.DebugInfoLib = "/usr/lib/debug"
+	config.Server.DebugInfoSrc = "/usr/src/debug"
+
+	// [Mixer]
+	config.Mixer.LocalBundleDir = filepath.Join(pwd, "local-bundles")
+	config.Mixer.LocalRPMDir = ""
+	config.Mixer.LocalRepoDir = ""
+
+	return nil
+}
+
+// CreateDefaultConfig creates a default builder.conf using the active
+// directory as base path for the variables values.
+func (config *MixConfig) CreateDefaultConfig(localrpms bool) error {
+	if !UseNewConfig {
+		return config.createLegacyConfig(localrpms)
+	}
+
+	if err := config.LoadDefaults(); err != nil {
+		return err
+	}
+
+	if localrpms {
+		pwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+
+		config.Mixer.LocalRPMDir = filepath.Join(pwd, "local-rpms")
+		config.Mixer.LocalRepoDir = filepath.Join(pwd, "local-yum")
+	}
+
+	filename, err := GetConfigPath("")
+	if err != nil {
+		return err
+	}
+
+	return config.SaveConfig(filename)
+}
+
+// SaveConfig saves the properties in MixConfig to a TOML config file
+func (config *MixConfig) SaveConfig(filename string) error {
+	if !UseNewConfig {
+		return errors.Errorf("SaveConfig can only be used with --new-config flag")
+	}
+
+	w, err := os.OpenFile(filename, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0666)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		_ = w.Close()
+	}()
+
+	enc := toml.NewEncoder(w)
+	return enc.Encode(config)
+}
+
+func (config *MixConfig) createLegacyConfig(localrpms bool) error {
+	if UseNewConfig {
+		return errors.Errorf("createLegacyConfig is not compatible with --new-config flag")
+	}
+
+	pwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	builderconf := filepath.Join(pwd, "builder.conf")
+
+	err = helpers.CopyFileNoOverwrite(builderconf, "/usr/share/defaults/bundle-chroot-builder/builder.conf")
```

Comment:
For future patch: We should really update this file itself to the new format so the user doesn't have to call convert on it later, and probably move it out of BCB and into mixer @matthewrsj

Reply:
I'd call the file under bcb deprecated. Does Clear devops still use this or something?

Reply:
I thought we were dropping that file since `mixer init --new-config` will generate the template in the new format already.

Reply:
Agreed to both. If I understand correctly, it's only used in the legacy code path, so rather than update the file we can just eventually remove it when `--new-config` is removed.